Packet++/header/SomeIpSdLayer.h

```diff
@@ -0,0 +1,998 @@
+		m_DataLen = sizeof(someipsdhdroptionsbase) - 1 + be16toh(pHdr->length);
+		m_OptType = optionType;
+	}
+
+	std::string getConfigurationString() const;
+
+	/**
+	 * get Configuration String
+	 * @return std::string
+	 */
+	std::string toString() const override;
+
+private:
+	SomeIpSdConfigurationStringOption *doCloneTo(IDataContainer *dataContainer, size_t offset) const override;
+
+	std::unique_ptr<SomeIpSdConfigurationStringOption> cloneTo(IDataContainer *dataContainer, size_t offset) const
+	{
+		return std::unique_ptr<SomeIpSdConfigurationStringOption>(doCloneTo(dataContainer, offset));
+	}
+
+	SomeIpSdConfigurationStringOption &operator=(const SomeIpSdConfigurationStringOption &rhs)
+	{
+		SomeIpSdOption::operator=(static_cast<const SomeIpSdOption &>(rhs));
+		return *this;
+	}
+};
+
+typedef std::unique_ptr<SomeIpSdOption> OptionPtr;
+typedef std::vector<OptionPtr> OptionsVec;
+typedef SomeIpSdOption *OptionRawPtr;
+typedef std::vector<OptionRawPtr> OptionsRawVec;
+
+/**
+ * @class SomeIpSdEntry
+ * Base class of the SOME/IP-SD entries. Cannot be instantiated.
+ */
+class SomeIpSdEntry
+{
+public:
+	friend class SomeIpSdLayer;
+
+	/**
+	 * Types of entries that can occur in SOME/IP-SD
+	 */
+	enum EntryType
+	{
+		/** Find Service */
+		FindService,
+		/** Offer Service */
+		OfferService,
+		/** Stop Offer Service */
+		StopOfferService,
+		/** Subscribe Eventgroup */
+		SubscribeEventgroup,
+		/** Stop Subscribe Eventgroup */
+		StopSubscribeEventgroup,
+		/** Subscribe Eventgroup Acknowledgment */
+		SubscribeEventgroupAck,
+		/** Subscribe Eventgroup Negative Acknowledgement */
+		SubscribeEventgroupNack
+	};
+
+	/**
+	 * Creates an entry of the right type and returns it as a base pointer. Used to create SomeIpSd
+	 * EntryType 1 objects.
+	 * @param[in] type Type to create
+	 * @param[in] serviceID ServiceID to use
+	 * @param[in] instanceID InstanceID to use
+	 * @param[in] majorVersion MajorVersion to use
+	 * @param[in] TTL TTL to use. Has to be 0 for all Stop* entry types
+	 * @param[in] minorVersion MinorVersion to use
+	 * @return std::unique_ptr<SomeIpSdEntry>
+	 */
+	static std::unique_ptr<SomeIpSdEntry> create(EntryType type, uint16_t serviceID, uint16_t instanceID,
+												 uint8_t majorVersion, uint32_t TTL, uint32_t minorVersion);
+
+	/**
+	 * Creates an entry of the right type and returns it as a base pointer. Used to create SomeIpSd
+	 * EntryType 2 objects.
+	 * @param[in] type Type to create
+	 * @param[in] serviceID ServiceID to use
+	 * @param[in] instanceID InstanceID to use
+	 * @param[in] majorVersion MajorVersion to use
+	 * @param[in] TTL TTL to use. Has to be 0 for all Stop* entry types
+	 * @param[in] counter Counter value to use
+	 * @param[in] eventGroupID EventgroupId to use
+	 * @return std::unique_ptr<SomeIpSdEntry>
+	 */
+	static std::unique_ptr<SomeIpSdEntry> create(EntryType type, uint16_t serviceID, uint16_t instanceID,

```

Comment:
I wonder if we need this static `create(...)` methods. The c'tor is public so the user can just call it to create an object, no?